ts/features/bonus/bpd/store/reducers/details/combiner.ts

```diff
@@ -49,31 +49,54 @@ export const bpdAllPeriodsWithAmountSelector = createSelector(
 );
 
 /**
- * A period is visible in the wallet if the bpd is enabled and the period is active OR
- * the period is closed and the transactionNumber > 0
+ * A period is visible in the wallet if the bpd is enabled AND the period is active OR
+ * the period is closed and the transactionNumber > 0 OR
+ * bpd is enabled AND the period is inactive (future) AND there are no active or closed period
+ * @param periodList
  * @param periodAmount
  * @param bpdEnabled
  */
 const isPeriodAmountWalletVisible = (
+  periodList: ReadonlyArray<BpdPeriodAmount>,
   periodAmount: BpdPeriodAmount,
   bpdEnabled: RemoteValue<boolean, Error>
 ) =>
   isReady(bpdEnabled) &&
   ((periodAmount.period.status === "Active" && bpdEnabled.value) ||
     (periodAmount.period.status === "Closed" &&
-      periodAmount.amount.transactionNumber > 0));
+      periodAmount.amount.transactionNumber > 0) ||
+    // All the periods are inactive
+    (periodList.every(p => p.period.status === "Inactive") &&
+      // This is the first inactive period
+      periodList.indexOf(periodAmount) === 0 &&
```

Comment:
This is a shallow equality: it checks about object reference.

```typescript

const inactivePeriodA: BpdPeriodAmount = {
  amount: zeroAmount,
  period: {
    ...inactivePeriod,
    startDate: new Date("2025-01-01"),
    awardPeriodId: 55 as AwardPeriodId
  }
};

const test = isPeriodAmountWalletVisible(
      [inactivePeriodA],
      inactivePeriodA,
      remoteReady(true)
    );

    const test2 = isPeriodAmountWalletVisible(
      [inactivePeriodA],
      { ...inactivePeriodA },
      remoteReady(true)
    );
```
`test -> true`
`test2 -> false`

A solution could be

```typescript
import _ from "lodash";
periodList.findIndex(o => _.isEqual(o, periodAmount)) === 0 
```